### ISSUE TITLE
Reduce the number of CI tests

### ISSUE THREAD

**Issue by certik** [symengine/symengine#2020]
Right now there are way too many CI tests, that increase the maintenance burden. Rather, one should test the representative build on each platform/compiler combination that we want to support. And then on the fastest runner (say Ubuntu 22) test the rest of the cmake options combinations.

Ideally 10 total tests or something like that.

**Comment by rikardn** [comment]
I agree. And having all tests run on GHA would also reduce the maintenance burden. Currently a handful still run on appveyor.

**Comment by lkeegan** [comment]
I agree with simplifying the CI tests - in the meantime @oliverlee suggested a nice solution to avoid having to update the list of required tests every time the CI is updated, by having a `check` CI job that passes if all other jobs pass and fails otherwise. Then only the `check` job needs to be required.
I made a PR here: https://github.com/symengine/symengine/pull/2049
